### PR TITLE
Added Section for displaying GitHub project on user profile

### LINKED ISSUE #2835: [FEATURE] Separate sections for projects
### A separate section to showcase the projects.
Currently, one can add links of their projects in `My Links` section through `links` field in their json data.

I think `My Links` is a more general space, one can figure out what to put there.

But having a `Projects` section can facilitate showcasing projects that one have worked on

### The data can look like
```json
 
    "projects": [
        {
            "title": "The name of the project",
            "techsUsed": [
                "ReactJS",
                "Typescript"
            ],
            "url": "url to the project if deployed",
            "githubRepo": " url to the repo if exists ",
            "contributors": []
        }
    ]

```

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by shyam0705**
## Fixes Issue
Closes #2835 

## Changes proposed
1)Added section to show GitHub project in the user profile
2)Modified documentation to guide users to include projects in the profile JSON file.

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots
image

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/3422"></a>

### CODE REVIEW

File: pages/api/users/[username].js [hunk 1]
```diff
@@ -167,6 +167,45 @@ export default async function handler(req, res) {
   const latestProfile = await Profile.findOne({ username });
   const links = await Link.find({ profile: latestProfile._id });
 
+  if (data.projects?.github) {
+    data.projects.github = await Promise.all(
+      data.projects.github.map(async (repo) => {
+        try {
+          const res = await fetch(`https://api.github.com/repos/${repo}`);
```
Comment (eddiejaoude) [inline pages/api/users/[username].js:174]: We will hit the GitHub rate limit very quickly, we need to think how to cache these

**Comment by eddiejaoude** [comment]
Looks interesting 👍  please add a screenshot to the pull request description.

I think this needs a bit more though so we don't hit the rate limit of GitHub API and then it stops working for people's profiles

File: pages/api/users/[username].js (hunk 1 above)
Comment (shyam0705) [reply pages/api/users/[username].js:174]: Don't have any prior experience in catching api calls but will look into this.

**Comment by shyam0705** [comment]
> Looks interesting 👍 please add a screenshot to the pull request description.
> 
> I think this needs a bit more though so we don't hit the rate limit of GitHub API and then it stops working for people's profiles

Screenshot is already there let me know if you mean something else or you are not able to see it.

**Comment by eddiejaoude** [comment]
Oh yes, sorry I see the screenshot 👍 

Now we have 5 tabs, does this fit correctly when all are being used?

**Comment by eddiejaoude** [comment]
There has been no activity for a while on this, so I will close.

Plus this can be looked into further once forms have been completed as we don't want to add anymore data to the json files https://github.com/EddieHubCommunity/LinkFree/pull/6998

**Pull request closed without merging**